run.sh

```diff
@@ -1,15 +1,58 @@
 #!/bin/bash
 set -e
 set -x
-export DEVPI_SERVERDIR=/mnt
-export DEVPI_CLIENTDIR=/tmp/devpi-client
-[[ -f $DEVPI_SERVERDIR/.serverversion ]] || initialize=yes
-
-devpi-server --start --host 0.0.0.0 --port 3141
-if [[ $initialize = yes ]]; then
-  devpi use http://localhost:3141
-  devpi login root --password=''
-  devpi user -m root password="${DEVPI_PASSWORD}"
-  devpi index -y -c public pypi_whitelist='*'
-fi
-tail -f $DEVPI_SERVERDIR/.xproc/devpi-server/xprocess.log
+
+# Default environment if none is provided
+: ${DEVPI_SERVERIP:=127.0.0.1}
+: ${DEVPI_SERVERPORT:=3143}
+: ${DEVPI_SERVERDIR:=/mnt}
+: ${DEVPI_CLIENTDIR:=/tmp/devpi-client}
+: ${DEVPI_CLIENTPASSWORD:=""}
+: ${DEVPI_LDAPCONFILE:=""}
+
+: ${NGINX_IP:=0.0.0.0}
+: ${NGINX_PORT:=3141}
+: ${NGINX_ENABLETLS:=false}
+: ${NGINX_TLSPORT:=3142}
+: ${NGINX_TLSPATH:=/opt/nginx/certs/default}
+: ${NGINX_TLSCERT:=devpi.crt}
+: ${NGINX_KEY:=devpi.key}
+export DEVPI_SERVERIP DEVPI_SERVERPORT DEVPI_SERVERDIR DEVPI_CLIENTDIR DEVPI_CLIENTPASSWORD DEVPI_LDAPCONFILE NGINX_IP NGINX_PORT NGINX_ENABLETLS NGINX_TLSPORT NGINX_TLSPATH NGINX_TLSCERT NGINX_KEY
+
+function start_devpi_server () {
+    # Start the server
+    devpi-server --start --host ${DEVPI_SERVERIP} --port ${DEVPI_SERVERPORT} &> stdout
+    
+    # Configure the default client if necessary
+    if [[ ! -f ${DEVPI_SERVERDIR}/.serverversion ]]; then
+      devpi use http://${DEVPI_SERVERIP}:${DEVPI_SERVERPORT}
+      devpi login root --password=""
+      devpi user -m root password="${DEVPI_CLIENTPASSWORD}"
+      devpi index -y -c public pypi_whitelist='*'
+    fi
+}
+
+function start_nginx () {
+    nginx_template_file=/opt/nginx/nginx-tpl.conf
```

Comment:
It could be good to make nginx_template_file a setting variable with default value `/opt/nginx/nginx-tpl.conf`